### ISSUE TITLE
Nested prop mutation is not flagged

### ISSUE THREAD

**Issue by sstern6** [vuejs/eslint-plugin-vue#1828]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.11.x
- **eslint-plugin-vue version:** 8.5.x
- **Node version:** 17.5.0
- **Operating System:** OSX

**Please show your full configuration:**

```json5
const path = require('path');

module.exports = {
  root: true,
  env: {
    browser: true,
    node: true,
    jest: true,
  },
  extends: [
    'plugin:vue/essential',
    'plugin:vue/recommended',
    'plugin:vue-a11y/base',
    'airbnb-base',
  ],
};

```

**What did you do?**	

```vue
export default {
  props: {
    thing: {
      type: Object,
      required: true,
    },
   mounted() {
      this.thing.nested = true;
   }
}

```

**What did you expect to happen?**
Error

**What actually happened?**
No error

**Repository to reproduce this issue**

**Comment by ota-meshi** [comment]
Duplicate of #1314

**Comment by sstern6** [comment]
@ota-meshi thanks for pointing this out. 

Curious on why nested prop mutation would be allowed but regular prop mutation isnt.

It would catch the same issue and i personally think it should be guarded against. 

WDYT?

Is there a way to introduce a strict flag you can toggle?
